lib/nuget-parser/nuspec-parser.ts

```diff
@@ -8,80 +8,126 @@ const debug = debugModule('snyk');
 
 const targetFrameworkRegex = /([.a-zA-Z]+)([.0-9]+)/;
 
-export async function parseNuspec(dep, targetFramework) {
-  return Promise.resolve()
-    .then(() => {
-      const nupkgPath = path.resolve(
-        dep.path,
-        dep.name + '.' + dep.version + '.nupkg',
+async function loadNuspecFromAsync(dep) {
+  const nupkgPath = path.resolve(
+    dep.path,
+    dep.name + '.' + dep.version + '.nupkg',
+  );
+
+  //just in case, does the code expect null to be returned on any error here?
+  if (!fs.existsSync(nupkgPath)) return null;
+
+  const nupkgData = fs.readFileSync(nupkgPath);
+  const nuspecZipData: any = await JSZip.loadAsync(nupkgData);
+
+  const nuspecFiles = Object.keys(nuspecZipData.files).filter(file => {
+    return path.extname(file) === '.nuspec';
+  });
+
+  if (!nuspecFiles) {
+    debug(`failed to read nupkg file from: ${nupkgPath}`);
+    return null;
+  }
+
+  if (!nuspecZipData) {
+    debug(`failed to open nupkg file as an archive from: ${nupkgPath}`);
+    return null;
+  }
+
+  return await nuspecZipData.files[nuspecFiles[0]].async('text');
```

Comment:
If we're going to use the first item, please replace the `.filter` usage with `.find`
If that's a mistake then please fix it :)